rfc-012-makefile-standards.md

```diff
@@ -0,0 +1,64 @@
+# Makefile standards
+
+## Summary
+
+A set of standards for Makefiles across all projects
+
+## Problem
+
+Having a standard set of commands that can be run across all projects for common tasks
+allows people to get projects up and running without having to spend time looking into
+how things are set up to know what specific commands to run.
+
+
+## Proposal
+
+All projects MUST have a Makefile.
+
+All projects must have relevant [`PHONY` targets](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html)
+
+**All** projects MUST have the commands:
+- `make test`
+  - Runs the tests for the project
+  - This SHOULD run the tests in a 'watch' mode (e.g. [Guard](https://github.com/guard/guard))
+- `make setup`
+  - Runs the setup for the project
+  - E.g. Runs `docker-compose build`
+
+If the project is a **web application** it MUST have the commands:
+- `make serve` 
+  - Runs the development version of the application, e.g. Running `docker-compose up`
+
+If the project is an application which needs compiling for deployment it MUST have the commands:
+- `make build`
+  - Builds the application for production
+
+If the project runs in **Docker** - it MUST have the commands:
+- `make docker-build` 
+  - Builds the Docker image for the project
+- `make docker-down`
```

Comment:
Potentially have a docker-agnostic target too, that could reference this, such as `make stop` - which stops any running processes for the project.